src/server/src/main/java/io/cassandrareaper/jmx/JmxConnectionFactory.java

```diff
@@ -171,37 +165,37 @@ public final HostConnectionCounters getHostConnectionCounters() {
     return accessibleDatacenters;
   }
 
-  public Optional<JmxCredentials> getJmxCredentialsForCluster(String clusterName) {
-    Optional<JmxCredentials> jmxCreds = Optional.ofNullable(jmxAuth);
-    if (jmxCredentials != null && jmxCredentials.containsKey(clusterName)) {
-      jmxCreds = Optional.of(jmxCredentials.get(clusterName));
-    }
+  public JmxCredentials getJmxCredentialsForCluster(Node node) {
+    JmxCredentials nodeCredentials = node.getJmxCredentials().orElse(jmxAuth);
+
+    if (nodeCredentials == null && jmxCredentials != null) {
+
+      if (jmxCredentials.containsKey(node.getClusterName())) {
+        nodeCredentials = jmxCredentials.get(node.getClusterName());
+      } else if (jmxCredentials.containsKey(Cluster.toSymbolicName(node.getClusterName()))) {
+        // As clusters get stored in the database with their "symbolic name" we have to look for that too
+        nodeCredentials = jmxCredentials.get(Cluster.toSymbolicName(node.getClusterName()));
+      }
 
-    // As clusters get stored in the database with their "symbolic name" we have to look for that too
-    if (jmxCredentials != null && jmxCredentials.containsKey(Cluster.toSymbolicName(clusterName))) {
-      jmxCreds = Optional.of(jmxCredentials.get(Cluster.toSymbolicName(clusterName)));
     }
 
-    return jmxCreds;
+    return nodeCredentials == null ? JmxCredentials.builder().build() : nodeCredentials;
```

Comment:
can we use a dedicated instance for no credentials instead please. eg `JmxCredentials.NO_CREDENTIALS`

(( Such a class might then be able to `throw UnsupportedOperation()` in its getters, rather than us having to support nullable getters. ))

Reply:
I took a slightly different path here and switched getJmxCredentialsForCluster(Node) to return an Optional<JmxCredentials>.  Doing this made it easier for the caller to do an .isPresent() check to see if there were credentials or not.  Please let me know if this approach works for you.